pandas/core/missing.py

```diff
@@ -82,16 +83,24 @@ def clean_fill_method(method, allow_nearest=False):
 
 
 def clean_interp_method(method, **kwargs):
+    import scipy
+    scipy_version = scipy.__version__
     order = kwargs.get('order')
     valid = ['linear', 'time', 'index', 'values', 'nearest', 'zero', 'slinear',
              'quadratic', 'cubic', 'barycentric', 'polynomial', 'krogh',
-             'piecewise_polynomial', 'pchip', 'akima', 'spline']
+             'piecewise_polynomial', 'pchip', 'akima', 'spline',
+             'from_derivatives']
     if method in ('spline', 'polynomial') and order is None:
         raise ValueError("You must specify the order of the spline or "
                          "polynomial.")
     if method not in valid:
         raise ValueError("method must be one of {0}."
                          "Got '{1}' instead.".format(valid, method))
+    # compat GH12887
+    if method == 'piecewise_polynomial' and LooseVersion(scipy_version) > \
+            LooseVersion('0.17'):
+        raise ValueError("Method '{0}' is deprecated for Scipy > 0.17. "
+                         "Use 'from_derivatives' instead".format(method))
```

Comment:
I raise a ValueError here, however it is isn't propagated back up to the calling function `interpolate` from `pandas/core/internals.py`, so open to suggestions assuming this approach is correct to handling it.